frontend/src/pages/profile/components/user-activity/user-activity.tsx

```diff
@@ -0,0 +1,88 @@
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { type UserActivityResponseDto } from '~/packages/users/users.js';
+
+import {
+  getActivityBreakpoint,
+  getUniqueMonths,
+} from '../../libs/helpers/helpers.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  userActivity: UserActivityResponseDto[];
+};
+
+const UserActivity: React.FC<Properties> = ({ userActivity }) => {
+  const uniqueMonths: string[] = getUniqueMonths(userActivity);
+
+  const activityStatistic = userActivity.map((activity) => {
+    const { count, date } = activity;
+    const activityBreakpoint = getActivityBreakpoint(count);
+
+    return (
+      <span
+        title={date}
+        key={date}
+        className={getValidClassNames(
+          styles.activityItem,
+          styles[activityBreakpoint],
+        )}
+      ></span>
+    );
+  });
+
+  return (
+    <>
+      {Boolean(userActivity.length) && (
```

Comment:
Explicit is always better than implicit 
```suggestion
      {userActivity.length > 0 && (
```